### PR TITLE
Select multiple envelopes at once using shift

### PR DESCRIPTION
**Pull request by st3iny**
Select multiple envelopes at once using the shift + click shortcut.

shift-select-multiple

### CODE REVIEW

**Review by GretaD — APPROVED** [review]
Tested and it works, but i think the select feature is broken on master, because if you see also at your gif, you select some read messages and the action button says mark as read.

File: src/components/EnvelopeList.vue [hunk 1]
```diff
@@ -92,14 +92,15 @@
 				class="icon-loading-small"
 				:class="{refreshing: refreshing}" />
 			<Envelope
-				v-for="env in envelopes"
+				v-for="(env, index) in envelopes"
 				:key="env.databaseId"
 				:data="env"
 				:mailbox="mailbox"
-				:selected="isEnvelopeSelected(envelopes.indexOf(env))"
+				:selected="isEnvelopeSelected(index)"
```
Comment (skjnldsv) [inline src/components/EnvelopeList.vue:99]: WHat happens if a new message arrives? Index will shift?
Is index the correct keyword here or is this a uuid?

Comment (StCyr) [reply src/components/EnvelopeList.vue:99]: As far as I remember:

I don't think we manage it yet when new messages arrive.
And, yes it's an index, not a message id.... Maybe not the wiser choice though

Comment (ChristophWurst) [reply src/components/EnvelopeList.vue:99]: it's the old issue from https://github.com/nextcloud/mail/pull/2619

time to fix this in a follow-up PR :wink:

**Comment by ChristophWurst** [comment]
> Tested and it works, but i think the select feature is broken on master, because if you see also at your gif, you select some read messages and the action button says mark as read.

Is that a blocker for this PR or can we merge?

**Review by StCyr — APPROVED** [review]
Looks good and work 👍

**Pull request merged**